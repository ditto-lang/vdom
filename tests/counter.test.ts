import { getByText, within } from "@testing-library/dom";
import { main } from "../dist/Example_Counter";

function renderCounter() {
  const root = document.createElement("div");
  const container = document.createElement("div");
  root.appendChild(container);
  main(container)();
  return root;
}

test("clicking the buttons", async () => {
  const container = renderCounter();

  const increment = () => within(container).getByText("+").click();
  const decrement = () => within(container).getByText("-").click();

  increment();
  increment();
  increment();
  decrement();

  expect(container).toMatchSnapshot();
});
