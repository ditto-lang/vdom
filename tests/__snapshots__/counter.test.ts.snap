// Vitest Snapshot v1

exports[`clicking the buttons 1`] = `
<div>
  <div>
    <button>
      -
    </button>
    2
    <button>
      +
    </button>
  </div>
</div>
`;
